core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -387,14 +389,24 @@ private void deleteRemovedMetadataFiles(TableMetadata base, TableMetadata metada
       // the log, thus we don't include metadata.previousFiles() for deletion - everything else can
       // be removed
       removedPreviousMetadataFiles.removeAll(metadata.previousFiles());
-      Tasks.foreach(removedPreviousMetadataFiles)
-          .noRetry()
-          .suppressFailureWhenFinished()
-          .onFailure(
-              (previousMetadataFile, exc) ->
-                  LOG.warn(
-                      "Delete failed for previous metadata file: {}", previousMetadataFile, exc))
-          .run(previousMetadataFile -> io().deleteFile(previousMetadataFile.file()));
+      if (io() instanceof SupportsBulkOperations) {
+        ((SupportsBulkOperations) io())
+            .deleteFiles(
+                Iterables.transform(
+                    removedPreviousMetadataFiles, TableMetadata.MetadataLogEntry::file));
+      } else {
+        LOG.warn(
+            "IO {} does not support bulk operations. Using non-bulk deletes.",
+            io().getClass().getName());
```

Comment:
imo don't think we need the log, at least I don't think warn level is correct since it's not really indicative of any problems.

Reply:
Yeah make sense, dropped as suggested